vm/factory/systemSCFactory.go

```diff
@@ -1,47 +1,81 @@
 package factory
 
 import (
-	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/core/check"
 	"github.com/ElrondNetwork/elrond-go/vm"
 	"github.com/ElrondNetwork/elrond-go/vm/systemSmartContracts"
 )
 
 type systemSCFactory struct {
-	systemEI          vm.SystemEI
-	validatorSettings process.ValidatorSettingsHandler
+	systemEI          vm.ContextHandler
+	validatorSettings vm.ValidatorSettingsHandler
+	sigVerifier       vm.MessageSignVerifier
 }
 
 // NewSystemSCFactory creates a factory which will instantiate the system smart contracts
 func NewSystemSCFactory(
-	systemEI vm.SystemEI,
-	validatorSettings process.ValidatorSettingsHandler,
+	systemEI vm.ContextHandler,
+	validatorSettings vm.ValidatorSettingsHandler,
+	sigVerifier vm.MessageSignVerifier,
 ) (*systemSCFactory, error) {
-	if systemEI == nil || systemEI.IsInterfaceNil() {
+	if check.IfNil(systemEI) {
 		return nil, vm.ErrNilSystemEnvironmentInterface
 	}
-	if validatorSettings == nil || validatorSettings.IsInterfaceNil() {
+	if check.IfNil(validatorSettings) {
 		return nil, vm.ErrNilEconomicsData
 	}
+	if check.IfNil(sigVerifier) {
+		return nil, vm.ErrNilMessageSignVerifier
+	}
 
 	return &systemSCFactory{
 		systemEI:          systemEI,
-		validatorSettings: validatorSettings}, nil
+		validatorSettings: validatorSettings,
+		sigVerifier:       sigVerifier}, nil
 }
 
 // Create instantiates all the system smart contracts and returns a container
 func (scf *systemSCFactory) Create() (vm.SystemSCContainer, error) {
 	scContainer := NewSystemSCContainer()
 
-	sc, err := systemSmartContracts.NewStakingSmartContract(
+	staking, err := systemSmartContracts.NewStakingSmartContract(
 		scf.validatorSettings.StakeValue(),
 		scf.validatorSettings.UnBoundPeriod(),
 		scf.systemEI,
+		AuctionSCAddress,
 	)
 	if err != nil {
 		return nil, err
 	}
 
-	err = scContainer.Add(StakingSCAddress, sc)
+	err = scContainer.Add(StakingSCAddress, staking)
+	if err != nil {
+		return nil, err
+	}
+
+	args := systemSmartContracts.ArgsStakingAuctionSmartContract{
+		MinStakeValue:    scf.validatorSettings.StakeValue(),
```

Comment:
Why not working directly with validatorSettingsHandler? Why do we need to copy all its values? This will not work if, for example, the values held by validatorSettings change during execution (from one epoch to another).

Reply:
done